### DISCUSSION TITLE
git buffer functions deprecated

### DISCUSSION THREAD

**Discussion by lya001** [libgit2/libgit2#6364] [General]
Hi,

So in our codebase, we need to copy a string into a git_buf in the function:

typedef int GIT_CALLBACK(git_merge_driver_apply_fn)(
	git_merge_driver *self,
	const char **path_out,
	uint32_t *mode_out,
	git_buf *merged_out,
	const char *filter_name,
	const git_merge_driver_source *src);

This used to be done by calling 'git_buf_set'. However, it is now deprecated.

My workaround is now to set it with "a->ptr = strdup(b);" where a is the git_buf and b is the string.

However, this is a bit 'asymmetric' as the memory is now assigned by the caller(us) but freed by libgit2 when calling "git_buf_dispose".

Does anyone have a better suggestion for how this could be done? Thanks!

**Comment by ethomson** [comment]
Hi @lya001 - if you're using `git_merge_driver_apply_fn` then that function is also deprecated. So I would encourage you to either move over to the streaming filter application (meaning that you don't need to worry about `git_buf_set`) or keep using the deprecated APIs. We'll remove those functions at some point but not in the next minor release. Possibly for libgit2 2.0 but certainly no sooner.

**Reply by lya001** [reply]
Thanks! I see `git_filter_apply_fn` is deprecated and there is `git_filter_stream_fn` for data filtering. But actually I think for merging `git_merge_driver_apply_fn` is not deprecated and it has a `git_buf *` parameter.
